packages/cli/test/integration/evaluation/test-definitions.api.test.ts

```diff
@@ -0,0 +1,296 @@
+	});
+
+	test('should return error if name is empty', async () => {
+		const resp = await authOwnerAgent.post('/evaluation/test-definitions').send({
+			name: '',
+			workflowId: workflowUnderTest.id,
+		});
+
+		expect(resp.statusCode).toBe(400);
+		expect(resp.body.message).toBe('Test definition name must be 1 to 255 characters long.');
+	});
+
+	test('should return error if user has no access to the workflow', async () => {
+		const resp = await authOwnerAgent.post('/evaluation/test-definitions').send({
+			name: 'test',
+			workflowId: otherWorkflow.id,
+		});
+
+		expect(resp.statusCode).toBe(400);
+		expect(resp.body.message).toBe('User does not have access to the workflow');
+	});
+
+	test('should return error if user has no access to the evaluation workflow', async () => {
+		const resp = await authOwnerAgent.post('/evaluation/test-definitions').send({
+			name: 'test',
+			workflowId: workflowUnderTest.id,
+			evaluationWorkflowId: otherWorkflow.id,
+		});
+
+		expect(resp.statusCode).toBe(400);
+		expect(resp.body.message).toBe('User does not have access to the evaluation workflow');
+	});
+});
+
+describe('PATCH /evaluation/test-definitions/:id', () => {
+	test('should update test definition', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.patch(`/evaluation/test-definitions/${newTest.id}`).send({
+			name: 'updated-test',
+		});
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.name).toBe('updated-test');
+	});
+
+	test('should update test definition with evaluation workflow', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.patch(`/evaluation/test-definitions/${newTest.id}`).send({
+			name: 'updated-test',
+			evaluationWorkflowId: evaluationWorkflow.id,
+		});
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.name).toBe('updated-test');
+		expect(resp.body.data.evaluationWorkflowId).toBe(evaluationWorkflow.id);
+	});
+
+	test('should return error if user has no access to the evaluation workflow', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.patch(`/evaluation/test-definitions/${newTest.id}`).send({
+			name: 'updated-test',
+			evaluationWorkflowId: otherWorkflow.id,
+		});
+
+		expect(resp.statusCode).toBe(400);
+		expect(resp.body.message).toBe('User does not have access to the evaluation workflow');
+	});
+
+	test('should ignore the provided workflowId', async () => {
```

Comment:
Personally I'd like this to return 400 if you are trying to change something you are not allowed to. Or is there a reason for doing it this way?

Reply:
I changed this to fail on request body validation, it will respond with 400 now